astropylibrarian/algolia/records.py

```diff
@@ -1,110 +1,314 @@
+    """A Pydantic model for an Learn Astropy record in Algolia."""
+
+    objectID: str = Field(description="Unique identifier for this record.")
+
+    index_epoch: UUID4 = Field(
+        description=(
+            "Indexing epoch for this root_url. All records with the same "
+            "root_url are ingested at the same time using the same "
+            "index_epoch value. This enables old records, with a "
+            "different index_epoch value to be deleted."
+        )
+    )
+
+    root_summary: Optional[str] = Field(
+        description=(
+            "Short summary of the content corresponding to the root_url."
+        )
+    )
+
+    content_type: ContentType = Field(description="Content type.")
+
+    url: HttpUrl = Field(
+        description=(
+            "The most-specific URL for this record. Compared to the base url "
+            "this URL may contain things like fragments in the URL."
+        ),
+    )
+
+    root_url: HttpUrl = Field(
+        description=(
+            "URL of the document project's root page. For multi-page sites "
+            "this corresponds to the site's homepage. For single-page sites "
+            "this corresponds to the page's URL and is the same as "
+            "``baseUrl``."
+        ),
+    )
+
+    root_title: str = Field(
+        description=(
+            "Title of the documentation project. For single-page sites "
+            "This is the same as ``h1``."
+        ),
+    )
+
+    base_url: HttpUrl = Field(
+        description=(
+            "The base URL of the page, without fragments, parameters, "
+            "queries, etc."
+        ),
+    )
+
+    h1: str = Field(description="The title.")
 
+    h2: Optional[str] = Field(description="The second-level heading.")
 
-@dataclass
-class TutorialSectionRecord:
-    """An Algolia record for learn.astropy tutorial sections."""
+    h3: Optional[str] = Field(description="The third-level heading.")
 
-    section: "Section"
-    """The underlying Section content of the record
-    (`astropylibrarian.reducers.utils.Section`).
-    """
+    h4: Optional[str] = Field(description="The fourth-level heading.")
 
-    tutorial: "ReducedTutorial"
-    """The reduced tutorial page that this record is associated with
-    (`astropylibrarian.reducers.tutorial.ReducedTutorial`).
-    """
+    h5: Optional[str] = Field(description="The fifth-level heading.")
 
-    keyworddb: KeywordDb = field(default_factory=KeywordDb.load)
-    """Keyword database."""
+    h6: Optional[str] = Field(description="The sixth-level heading.")
 
-    @property
-    def object_id(self) -> str:
-        """The objectID of the record.
+    importance: int = Field(
+        description="The importance of the record, corresponding to the "
+        "hierarchical section level"
+    )
 
-        This is computed based on the URL and section heading hierarchy.
+    content: str = Field(description="The plain text content of the record.")
```

Comment:
Just to be clear: here "plain text" is actually HTML?

Reply:
It really is the unstyled text; effectively all the content that goes inside HTML tags. How we get this for web pages is with [lxml.html's text_content() method](https://lxml.de/lxmlhtml.html). The reason we only want to track plain text content is that that's what Algolia is searching against, so we don't want all that markup in this instance.